model/patch/patch.go

```diff
@@ -326,5 +331,39 @@ func (p *Patch) RemoveModulePatch(moduleName string) error {
 }
 
 func (p *Patch) IsGithubPRPatch() bool {
-	return p.GithubPatchData.PRNumber != 0
+	return p.GithubPatchData.BaseOwner != ""
+}
+
+func (p Patch) IsPRMergePatch() bool {
+	return p.GithubPatchData.MergeCommitSHA != ""
+}
+
+func MakeMergePatch(pr *github.PullRequest, projectID string) (*Patch, error) {
+	u, err := user.GetPatchUser(pr.GetUser().GetID())
```

Comment:
user can probably be nil. 

we should also make sure that pr.GetUser() can't return nil...